src/libre/parser.act

```diff
@@ -413,6 +416,23 @@
 		return NULL;
 	}
 
+	static struct fsm *
+	negate(struct fsm *class, const struct fsm_options *opt)
+	{
+		struct fsm *any;
+
+		any = class_any_fsm(opt);
+
+		if (any == NULL || class == NULL) {
+			if (any) fsm_free(any);
+			if (class) fsm_free(class);
+		};
+
+		class = fsm_subtract(any, class);
+
+		return class;
+	}
```

Comment:
Could this be shared with `<invert-group>`?

Reply:
it is now!